radix-engine-interface/src/blueprints/epoch_manager/invocations.rs

```diff
@@ -50,9 +51,62 @@ pub const EPOCH_MANAGER_NEXT_ROUND_IDENT: &str = "next_round";
+    /// reported `round + 1`. Such gaps are considered "round leader's fault" and are penalized
+    /// on emission, according to leader reliability statistics (see `LeaderProposalMissHistory`).
     pub round: u64,
+
+    /// A captured history of leader proposal misses since the previous reported round.
+    pub leader_proposal_misses: LeaderProposalMissHistory,
+}
+
+impl EpochManagerNextRoundInput {
+    /// Creates a "next round" input for a regular (happy-path, in terms of consensus) round
+    /// progression, i.e. no missed proposals, no fallback rounds.
+    /// Please note that the round number passed here should be an immediate successor of the
+    /// previously reported round.
+    pub fn successful(consecutive_round: u64) -> Self {
+        Self {
+            round: consecutive_round,
+            leader_proposal_misses: LeaderProposalMissHistory {
+                gap_round_leaders: Vec::new(),
+                current_round_fallback_leader: None,
+            },
+        }
+    }
 }
 
+#[derive(Debug, Clone, Eq, PartialEq, Sbor)]
+pub struct LeaderProposalMissHistory {
+    /// The validators which were leaders of the "gap" rounds (i.e. those that were not reported to
+    /// the epoch manager since the previous call; see `EpochManagerNextRoundInput::round`).
+    /// This list will contain exactly `current_call.round - previous_call.round - 1` elements; in
+    /// theory, this makes `EpochManagerNextRoundInput::round` field redundant (i.e. computable),
+    /// but this relation can be used for an extra consistency check.
+    /// The validators on this list should be penalized during emissions at the end of the current
+    /// epoch.
+    pub gap_round_leaders: Vec<ValidatorIndex>,
+
+    /// An indication of the current round operating in a "fallback" mode (i.e. indicating a fault
+    /// of the current leader).
+    /// A `None` denotes a regular round (not a fallback).
+    /// A `Some(current_leader)` specifies the current round's leader and indicates that current
+    /// round is a fallback one (and that validator should be penalized during emissions, same as
+    /// `gap_round_leaders`).
+    /// Please note that this effect is independent from the gap rounds (i.e. the value of this
+    /// field does not affect the invariants of `gap_round_leaders` or the round numbers).
+    pub current_round_fallback_leader: Option<ValidatorIndex>,
+}
+
+/// An index of a specific validator within the current validator set.
+/// To be exact: a `ValidatorIndex` equal to `k` references the `k-th` element returned by the
+/// iterator of `BTreeMap<ComponentAddress, Validator>`.
+/// This uniquely identifies the validator, while being shorter than `ComponentAddress` (we do care
+/// about the constant factor of the space taken by `LeaderProposalMissHistory` under prolonged
+/// liveness break scenarios).
+pub type ValidatorIndex = usize;
```

Comment:
We may even wish to consider using a u8 (8 times smaller than a usize = u64), but there are maybe bigger fish to fry (like the fact validators are storing every round's proof at the moment).

Reply:
if we go this way (i.e. shave any bytes off the very long validator ID) I agree we should go to the limit.
(I assume there is some business rule that ensures validatorSet.len < 256 and that we plan to hold it for a long time)